src/app/board/quest/quest.component.ts

```diff
@@ -1,11 +1,19 @@
+import { HttpClient } from '@angular/common/http';
 import { Component, Input } from '@angular/core';
 import { IQuest } from '../../shared/quest';
+import { QuestService } from 'src/app/shared/quest.service';
 
 @Component({
   selector: 'app-quest',
   templateUrl: './quest.component.html',
   styleUrls: ['./quest.component.scss']
 })
 export class QuestComponent {
+  constructor(private QuestService: QuestService) {}
   @Input() quest: IQuest;
+
+  toggleAccept(quest: IQuest) {
+    this.QuestService.updateQuest({...quest, isAccepted: !quest.isAccepted})
+    .subscribe(() => quest.isAccepted = !quest.isAccepted)
```

Comment:
It's not good that you mutate data from a kind of presentational component. You should have single source of truth and single source of changes to have predictable code.